pdp/handlers.go

```diff
@@ -909,15 +914,195 @@ func (p *PDPService) handleAddRootToProofSet(w http.ResponseWriter, r *http.Requ
 		return true, nil
 	}, harmonydb.OptionRetry())
 	if err != nil {
-		log.Errorw("Failed to insert into database", "error", err, "txHash", txHash.Hex(), "subroots", subrootInfoMap)
+		log.Errorw("Failed to insert into database", "error", err, "txHash", txHashLower, "subroots", subrootInfoMap)
 		http.Error(w, "Internal server error", http.StatusInternalServerError)
 		return
 	}
 
 	// Step 10: Respond with 201 Created
+	w.Header().Set("Location", path.Join("/pdp/proof-sets", proofSetIDStr, "roots/added", txHashLower))
 	w.WriteHeader(http.StatusCreated)
 }
 
+// handleGetRootAdditionStatus handles GET /pdp/proof-sets/{proofSetID}/roots/added/{txHash}
+func (p *PDPService) handleGetRootAdditionStatus(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	// Step 1: Verify that the request is authorized using ECDSA JWT
+	serviceLabel, err := p.AuthService(r)
+	if err != nil {
+		http.Error(w, "Unauthorized: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	// Step 2: Extract parameters from the URL
+	proofSetIDStr := chi.URLParam(r, "proofSetID")
+	txHash := chi.URLParam(r, "txHash")
+
+	if proofSetIDStr == "" {
+		http.Error(w, "Missing proof set ID in URL", http.StatusBadRequest)
+		return
+	}
+	if txHash == "" {
+		http.Error(w, "Missing transaction hash in URL", http.StatusBadRequest)
+		return
+	}
+
+	// Convert proofSetID to uint64
+	proofSetID, err := strconv.ParseUint(proofSetIDStr, 10, 64)
+	if err != nil {
+		http.Error(w, "Invalid proof set ID format", http.StatusBadRequest)
+		return
+	}
+
+	// Clean txHash (ensure it starts with '0x' and is lowercase)
+	if !strings.HasPrefix(txHash, "0x") {
+		txHash = "0x" + txHash
+	}
+	txHash = strings.ToLower(txHash)
+
+	// Validate txHash is a valid hash
+	if len(txHash) != 66 { // '0x' + 64 hex chars
+		http.Error(w, "Invalid txHash length", http.StatusBadRequest)
+		return
+	}
+	if _, err := hex.DecodeString(txHash[2:]); err != nil {
+		http.Error(w, "Invalid txHash format", http.StatusBadRequest)
+		return
+	}
+
+	// Step 3: Verify proof set ownership
+	var proofSetService string
+	err = p.db.QueryRow(ctx, `
+		SELECT service
+		FROM pdp_proof_sets
+		WHERE id = $1
+	`, proofSetID).Scan(&proofSetService)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			http.Error(w, "Proof set not found", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "Failed to retrieve proof set: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if proofSetService != serviceLabel {
+		// Same response as not found to avoid leaking information
+		http.Error(w, "Proof set not found", http.StatusNotFound)
+		return
+	}
+
+	// Step 4: Query pdp_proofset_root_adds for this transaction
+	type RootAddInfo struct {
+		Root            string `db:"root"`
+		AddMessageIndex int    `db:"add_message_index"`
+		Subroot         string `db:"subroot"`
+		SubrootOffset   int64  `db:"subroot_offset"`
+		SubrootSize     int64  `db:"subroot_size"`
+		AddMessageOK    *bool  `db:"add_message_ok"`
```

Comment:
why do we use a pointer for reading `AddMessageOK ` but values for everything else ?

Reply:
What is the difference between `AddMessageOK =true` and `txStatus == "confirmed"` ?